app/services/sso/jwt.rb

```diff
@@ -24,6 +27,18 @@ def authenticated?
       self.user = User.current.presence || authenticate!
     end
 
+    def allowed_to?(action)
+      return false if @current_user.disabled?
+      return true if @current_user.admin?
+
+      required_permissions = Foreman::AccessControl.permissions_for_controller_action(action).map(&:name)
+      required_permissions.each do |permission|
+        return false unless @scope.include? permission.to_s
+      end
+
+      true
```

Comment:
I think the default case should be to prohibit an action, so this should return false here. Only if all the required permissions are set, access should be explicitly granted. Note, that you can use `&, | and -` on arrays to get the intersection, union and difference.